barretenberg/cpp/src/barretenberg/dsl/acir_format/ecdsa_constraints.cpp

```diff
@@ -92,13 +92,11 @@ void create_ecdsa_verify_constraints(typename Curve::Builder& builder,
         r[0] = field_ct::conditional_assign(predicate, r[0], field_ct(1)); // 0 < r < n
         s[0] = field_ct::conditional_assign(predicate, s[0], field_ct(1)); // 0 < s < n/2
 
-        // P is on the curve
-        typename Curve::AffineElement default_point(Curve::g1::one + Curve::g1::one);
-        // BIGGROUP_AUDITTODO: mutable accessor needed for conditional_assign(). Could add a conditional_assign method
-        // to biggroup or could just perform these operations on the underlying fields prior to constructing the
-        // biggroup element.
-        public_key.x() = Fq::conditional_assign(predicate, public_key.x(), default_point.x());
-        public_key.y() = Fq::conditional_assign(predicate, public_key.y(), default_point.y());
+        // P is on the curve. Use conditional_select to avoid needing mutable accessors.
+        // When predicate is true (valid signature), use the original public_key.
+        // When predicate is false (edge case), use a known good default point.
+        G1 default_point(Curve::g1::one + Curve::g1::one);
```

Comment:
Unrelated to your changes here but it seems like this particular choice of default point deserves a comment..

Reply:
added a comment